pootle/apps/pootle_statistics/views.py

```diff
@@ -17,20 +18,43 @@
 from pootle.core.utils.stats import (TOP_CONTRIBUTORS_CHUNK_SIZE,
                                      get_top_scorers_data)
 from pootle.core.views import PootleJSONMixin
+from pootle_app.models.directory import Directory
 from pootle_language.views import LanguageBrowseView
 from pootle_misc.util import ajax_required
 from pootle_project.views import ProjectBrowseView, ProjectsBrowseView
 from pootle_translationproject.views import TPBrowseStoreView, TPBrowseView
 
-from .forms import StatsForm
+from .forms import ContributorsForm, StatsForm
 
 
-class ContributorsJSONMixin(PootleJSONMixin):
+class StatsJSONMixin(PootleJSONMixin):
     @property
     def path(self):
         return self.kwargs["path"]
 
     def get_context_data(self, *args, **kwargs):
+        stats = self.stats
+
+        if (isinstance(self.object, Directory) and
+            'virtualfolder' in settings.INSTALLED_APPS):
+            stats['vfolders'] = {}
+
+            for vf_treeitem in self.object.vf_treeitems.iterator():
+                if self.request.user.is_superuser or vf_treeitem.is_visible:
```

Comment:
im noticing this is untested, and im wondering if we should move this bit to plugin as part of this refactoring